sticker-graph/sticker_graph/write_helper.py

```diff
@@ -19,18 +48,27 @@ def create_graph(config, model, args):
     tfconfig = tf.ConfigProto(gpu_options=gpuopts)
 
     with tf.Graph().as_default(), tf.Session(config=tfconfig) as session:
-        with tf.variable_scope("model", reuse=None):
-            model(config=config, shapes=shapes)
+        logdir = tf.placeholder(shape=[], name="logdir", dtype=tf.string)
+        summary_writer = _create_file_writer_generic_type(logdir)
+
+        with summary_writer.as_default(), tf.contrib.summary.always_record_summaries():
+            with tf.variable_scope("model", reuse=None):
```

Comment:
Just to be sure: does this `variable_scope` need to be scoped inside the summary_writer? Or could it be added to the conjunction above? (If so, that would be clearer.)

Reply:
Something like that?

```py
with summary_writer.as_default(), tf.contrib.summary.always_record_summaries(), tf.variable_scope("model",
                                                                                                  reuse=None):
    model(config=config, shapes=shapes)
    tf.group(tf.contrib.summary.summary_writer_initializer_op(),
             name="summary_init")
    serialized_graph = session.graph_def.SerializeToString()
    serialized_graph_tensor = tf.convert_to_tensor(serialized_graph)
    tf.contrib.summary.graph(serialized_graph_tensor, 0, name='graph_write')
```

It should work. I didn't go for it since it would place the summary ops inside the `model` scope. Having the 3 contexts in the same `with` also seems visually unclear. I like the original version as it seperates the summary stuff from the model related scope.

Reply:
Like that yes. Well, with these things I always worry that Tensorflow context managers introduce new variable scopes and the ops are renamed ;)

I see your point about summary op scopes. So, then a slightly different question: does the model variable scope need to be within or together with `summary_writer` at all? Can they be placed in parallel?

~~~python
with tf.variable_scope("model", reuse=None):
  # Model stuff

with summary_writer.as_default(), tf.contrib.summary.always_record_summaries():
 # Summary stuff
~~~

Or do the ops have to be created within the `summary_writer` context manager for it to pick up the graph nodes?

Reply:
I just tested it, it doesn't pick up the summary ops if the model is not created inside the scope. I did not experience any context manager related issues. I'd not expect these managers to mess with any variable scope.